wallet/src/ui/app/components/transactions-card/index.tsx

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+
+import AccountAddress from '_components/account-address';
+import ExplorerLink from '_components/explorer-link';
+import { ExplorerLinkType } from '_components/explorer-link/ExplorerLinkType';
+
+import type { ExecutionStatusType, TransactionKindName } from '@mysten/sui.js';
+
+import st from './TransactionsCard.module.scss';
+
+type TxnData = {
+    To?: string;
+    seq: number;
+    txId: string;
+    status: ExecutionStatusType;
+    txGas: number;
+    kind: TransactionKindName | undefined;
+    From: string;
+};
+const truncate = (txt: string, maxLength: number) => {
+    if (txt.length < maxLength + 3) {
+        return txt;
+    }
+    const beginningLength = Math.ceil(maxLength / 2);
+    const endingLength = maxLength - beginningLength;
+    return `${txt.substring(0, beginningLength)}...${txt.substring(
+        txt.length - endingLength
+    )}`;
+};
+
+function TransactionResult({ txresults }: { txresults: TxnData[] }) {
```

Comment:
Changing this to TransctionListItem or something and move the `map` functionality to a TransactionList where we can show error/loading states would make easier to reuse it maybe, and also use the useMiddleEllipsis hook